portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/PortletRequestMapper.java

```diff
@@ -99,22 +100,23 @@ else if (requestHandler instanceof ResourceReferenceRequestHandler) {
 		else if (requestHandler instanceof BookmarkablePageRequestHandler) {
 			url = encodeRenderUrl(url, true);
 		}
-		else if (requestHandler instanceof ListenerInterfaceRequestHandler) {
-			ListenerInterfaceRequestHandler listenerInterfaceRequestHandler = (ListenerInterfaceRequestHandler) requestHandler;
+		//added support for wicket ajax requets. The BookmarkableListenerInterfaceRequestHandler is associated with wicket ajax requests.
```

Comment:
I am sorry for the confusion because of the above comment. The right comment could be "added portlet handling for BookmarkableListenerInterfaceRequestHandler".